metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -279,21 +295,21 @@ final class RenameProvider(
     if (colonNotAllowed) {
       client.showMessage(forbiddenColonRename(name, newName))
     }
-    symbolIsLocal(symbol) && (!desc.isMethod || (!colonNotAllowed && !isForbidden))
+    symbolIsLocal(symbol, definitionPath) && (!desc.isMethod || (!colonNotAllowed && !isForbidden))
   }
 
-  private def symbolIsLocal(symbol: String): Boolean = {
+  private def symbolIsLocal(
+      symbol: String,
+      definitionPath: AbsolutePath
+  ): Boolean = {
 
-    def isFromWorkspace =
-      index
-        .definition(MSymbol(symbol))
-        .exists { definition =>
-          val isLocal = definition.path.isWorkspaceSource(workspace)
-          if (isLocal && definition.path.isJava) {
-            client.showMessage(javaSymbol(symbol.desc.name.value))
-          }
-          isLocal
-        }
+    def isFromWorkspace = {
+      val isLocal = definitionPath.isWorkspaceSource(workspace)
```

Comment:
for me `isLocal` has a meaning that symbol is `localXXX`(anonymous in a given file) and here we talk about `isInWorkspace` ?

Reply:
Changed to `isWorkspaceSymbol`